lib/HLSL/DxilGenerationPass.cpp

```diff
@@ -315,6 +353,16 @@ class DxilGenerationPass : public ModulePass {
 
     GenerateDxilOperations(M, handleMap);
 
+    if (NotOptimized || m_HasDbgInfo) {
```

Comment:
When optimization is enabled, local resource will be removed in early pass. This is only for no optimization or has debug info case.